### ISSUE TITLE
Bug: `flat config global ignores` should support `git-ignore-alike-syntax` to negate pattern

### ISSUE THREAD

**Issue by loynoir** [eslint/eslint#18184]
### Environment

Node version:  v21.7.0
npm version: 10.5.0
Local ESLint version: 
Global ESLint version:  v8.57.0 
Operating System: 

### What parser are you using?

Default (Espree)

### What did you do?

https://eslint.org/docs/latest/use/configure/configuration-files-new

> If ignores is used without any other keys in the configuration object, then the patterns act as global ignores. 

But currently, `flat config global ignores` does not support `git-ignore-alike-syntax` to negate pattern.

Thus, when project structure grows complexer, ugly hard code grows larger.

### What did you expect to happen?

Pretty config:

```js
export default [
  {
    ignores: ['!ignore42/a.js', '!src/']
  },
  {
    rules: {
      semi: 'error',
      'prefer-const': 'error'
    }
  }
]
```

Pretty config is working:

```sh
/path/to/reproduce/ignore42/a.js
  1:4  error  Missing semicolon  semi

/path/to/reproduce/src/a.js
  1:4  error  Missing semicolon  semi

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.

```

### What actually happened?

Pretty config:

```js
export default [
  {
    ignores: ['!ignore42/a.js', '!src/']
  },
  {
    rules: {
      semi: 'error',
      'prefer-const': 'error'
    }
  }
]
```

Pretty config is not working:

```sh
/path/to/reproduce/eslint.config.js
  11:2  error  Missing semicolon  semi

/path/to/reproduce/ignore1/a.js
  1:4  error  Missing semicolon  semi

/path/to/reproduce/ignore2/a.js
  1:4  error  Missing semicolon  semi

/path/to/reproduce/ignore42/a.js
  1:4  error  Missing semicolon  semi

/path/to/reproduce/ignore42/b.js
  1:4  error  Missing semicolon  semi

/path/to/reproduce/ignore42/c.js
  1:4  error  Missing semicolon  semi

/path/to/reproduce/src/a.js
  1:4  error  Missing semicolon  semi

✖ 7 problems (7 errors, 0 warnings)
  7 errors and 0 warnings potentially fixable with the `--fix` option.

```

Ugly config as workaround:

```js
export default [
  {
    ignores: ['eslint.config.js', 'ignore1/a.js', 'ignore2/a.js', 'ignore42/b.js', 'ignore42/c.js']
  },
  {
    rules: {
      semi: 'error',
      'prefer-const': 'error'
    }
  }
]
```

Ugly config is working:

```sh
/path/to/reproduce/ignore42/a.js
  1:4  error  Missing semicolon  semi

/path/to/reproduce/src/a.js
  1:4  error  Missing semicolon  semi

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.

```

### Link to Minimal Reproducible Example

https://github.com/loynoir/reproduce-eslint-18184

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @loynoir, thanks for the issue!

Flat config global ignores support negated patterns, and this works the same as .gitignore: negated patterns unignore previously ignored files and directories. This means that you have to first ignore something for negated patterns to have any effect. In other words, a negated pattern alone doesn't have an effect that anything that doesn't match it should be automatically ignored.

In your example, assuming that you'd like to ignore everything except `ignore42/a.js` and content of `src/`, global ignore patterns can be specified like this:

```js
export default [
  {
    ignores: [
      '**/*', // ignore everything
      '!ignore42/', // unignore `ignore42` directory but not its content
      '!ignore42/a.js', // unignore `ignore42/a.js`
      '!src/**' // unignore `src/` directory and its content
    ]
  },
  {
    rules: {
      semi: 'error',
      'prefer-const': 'error'
    }
  }
]
```
